tide-core/src/response.rs

```diff
@@ -7,12 +7,17 @@ pub type Response = http_service::Response;
 
 /// Serialize `t` into a JSON-encoded response.
 pub fn json<T: serde::Serialize>(t: T) -> Response {
-    // TODO: remove the `unwrap`
-    http::Response::builder()
-        .status(http::status::StatusCode::OK)
-        .header("Content-Type", "application/json")
-        .body(Body::from(serde_json::to_vec(&t).unwrap()))
-        .unwrap()
+    let mut res = http::Response::builder();
+
+    match serde_json::to_vec(&t) {
+        Ok(v) => res
+            .header("Content-Type", "application/json")
+            .body(Body::from(v)),
+        Err(e) => res
+            .status(http::status::StatusCode::INTERNAL_SERVER_ERROR)
+            .body(Body::from(e.to_string())),
```

Comment:
We probably shouldn't leak details to the client here; instead logging using `log::error` and returning an empty body seems like a better option (: